tools/genesis-gen/src/main.rs

```diff
@@ -0,0 +1,74 @@
+use std::num::NonZeroUsize;
+
+use clap::Parser;
+use color_eyre::Result;
+
+/// Generator main stuff
+mod gen;
+
+/// `RawGenesisBlock` generator.
+///
+/// Generated stuff:
+///
+/// - Domains, Accounts and Asset Definitions
+/// - Asset lifecycle actions (mints, burns, transfers)
+///
+/// TODO:
+///
+/// - Generate random (or not so) metadata
+#[derive(Parser, Debug)]
+pub struct CLI {
+    /// Minify output JSON
+    #[clap(long, short)]
+    minify: bool,
+    /// Print only genesis, without generated accounts data
+    #[clap(long, short = 'g')]
+    only_genesis: bool,
+    /// How many domains to generate
+    #[clap(long, short, default_value = "5")]
+    pub domains: usize,
```

Comment:
I would advise against using architecture-dependent types. Iroha does not support any. Better to explicitly use `u64` and `try_into` as needed, than to rely on something that can mismatch in the future.

Reply:
Using `u32` now.